plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/spi/impl/DefaultDiagramHelper.java

```diff
@@ -196,6 +197,40 @@ public Shape getLifelineBodyShape(View lifelineView) {
 		return new DeferredCreateCommand<>(Shape.class, editingDomain, parameters, shape);
 	}
 
+	@Override
+	public CreationCommand<Shape> createDestructionOccurrenceShape(Supplier<? extends MessageEnd> destruction,
+			Shape lifelineBody, int yPosition) {
+
+		// TODO: The Logical Model is required to have no dependencies on the diagram editor,
```

Comment:
I'm not sure I can follow this TODO. Can we get rid of it? If there is something to do, we should rather open a new issue for that.

Reply:
Every create method in the DefaultDiagramHelper has this TODO, so I basically added it for consistency, because I did not implement it in an different fashion as e.g. createExecutionShape.